### DISCUSSION TITLE
[Hackathon Use Case] - Chat Integration with Contextual Transformation on FDC3

### DISCUSSION THREAD

**Discussion by maoo** [finos/finos-pmcs#244] [Ideas]
# Enhance Chat Integration with Contextual Transformation on FDC3
_From @sarahstone-cosaic (Finsemble)_

## Business Problem
Facilitate communications between Buy-side and Sell-side by initiating a message to an existing chat group from an order in your order blotter. Use FDC3 intents to initiate a chat from an order on a blotter. Create a service to transform the “broker” or “instrument” on blotter to correct the chat name for that user. Potential to leverage Legend project to store and manage translations.

The buy-side would take advantage of options to chat for a given order to a variety of targets e.g. `internal execution desk`, `users supervisor` and `internal team`.

## Expected features

- Send order from blotter to correct chat room based on security or contra broker
- Multiplexing request to send to chat and open chart

## Useful apps, systems, projects, technologies, libraries and datasets

- [Legend](https://legend.finos.org)
- [FDC3](https://fdc3.finos.org)

**Comment by robmoffat** [comment]
"Improving the workflow around trading over chat"

**Comment by robmoffat** [comment]
@sarahstone-cosaic:  can you give us a bit more info about how Legend plays into this use-case?

**Reply by sarahaislinnstone** [reply]
I'm not super familiar with legend but the idea was that the data mappings could be housed and managed in legend

**Comment by TheJuanAndOnly99** [comment]
Closing discussion in preparation [for this years hackathon](https://www.finos.org/hosted-events/2024-10-25-citi-india-hackathon-supported-by-finos). Feel free to reopen and update if its still relevant.
